### PR TITLE
Changed  string prefixes used for mac and ip addresses to integers

### PR DESCRIPTION
**Pull request by pinsleepe**
@paulprozesky 
@brickZA

### CODE REVIEW

File: src/fxcorrelator_fengops.py [hunk 1]
```diff
@@ -34,18 +36,16 @@ def feng_initialise(corr):
...
-    macprefix = corr.configd['fengine']['10gbe_macprefix']
-    macbase = int(corr.configd['fengine']['10gbe_macbase'])
+    start_mac = int(Mac(corr.configd['fengine']['10gbe_start_mac']))
     # Set up shared board info
     boards_info = {}
     board_id = 0
     for f in corr.fhosts:
         macs = []
-        ips = []
+        # ips = [] #  not used
```
Comment (brickZA) [inline src/fxcorrelator_fengops.py]: Just remove it then, no need to keep the comment around

File: src/fxcorr.ini [hunk 2]
```diff
@@ -79,8 +78,7 @@ accumulation_len = 1526
 # Number of accumulations performed in the X engine cores
 xeng_accumulation_len = 256 
 # MAC, IP and port for 10Gbe cores
-10gbe_macprefix = 02:02:00:00:02:
-10gbe_macbase = 10
+10gbe_start_mac = 02:02:00:00:02:0a
 10gbe_start_ip = 10.100.0.110
 10gbe_start_port = 8888
```
Comment (brickZA) [inline src/fxcorr.ini]: Should this not also be changed to 10gbe_port?

File: src/filthost_fpga.py [hunk 3]
```diff
@@ -129,17 +131,18 @@ def setup_rx(self):
...
+        _ip_start = int(IpAddress(self._config['10gbe_start_ip']))
+
+        _mac_start = int(Mac(self._config['10gbe_start_mac']))
+        _mac_base = 0 + (self.board_id * _num_tengbes)
         for _gbe in self.tengbes:
-            this_mac = '%s%02x' % (_mac_prefix, _mac_base)
-            this_ip = '%s%d' % (_ip_prefix, _ip_base)
+            this_mac = _mac_start + _mac_base
+            this_ip = _ip_start + _ip_base
             _gbe.setup(mac=this_mac, ipaddress=this_ip,
```
Comment (brickZA) [inline src/filthost_fpga.py]: is gbe.setup() ok with the ints rather than strings as parameters?

File: src/fxcorrelator_fengops.py [hunk 4]
```diff
@@ -34,18 +36,16 @@ def feng_initialise(corr):
...
         for gbe in f.tengbes:
-            this_mac = '%s%02x' % (macprefix, macbase)
+            this_mac = start_mac + gbe
```
Comment (brickZA) [inline src/fxcorrelator_fengops.py]: Can gbe be used as a number? Is it not a 'gbe object' type thing?

File: src/dsimhost_fpga.py [hunk 5]
```diff
@@ -195,13 +195,9 @@ def setup_tengbes(self):
...
 
-        ip_bits = start_ip.split('.')
-        ipbase = int(ip_bits[3])
-        mac_bits = start_mac.split(':')
-        macbase = int(mac_bits[5])
         for ctr in range(num_tengbes):
-            mac = '%s:%d' % (':'.join(mac_bits[0:5]), macbase + ctr)
-            ip = '%s.%s.%s.%d' % (ip_bits[0], ip_bits[1], ip_bits[2], ipbase + ctr)
+            mac = str(tengbe.Mac(start_mac + ctr))
+            ip = str(tengbe.IpAddress(start_ip + ctr))
```
Comment (pinsleepe) [inline src/dsimhost_fpga.py]: So here I set mac and ip to strings for gbe.setup(). It is a small change in the rest of the files but I was thinking maybe it's better to change gbe.setup() so it accepts ints?

Comment (brickZA) [inline src/dsimhost_fpga.py]: I'd be happy.

Just use the tengbe.IpAddress and tengbe.Mac classes to parse the input parameters in gbe.setup(), I think that will take care of both strings and ints for you.

**Comment by brickZA** [comment]
This does not merge cleanly anymore. Could you merge devel into this branch manually and resolve the conflicts?

**Pull request closed without merging**